capsules/src/driver.rs

```diff
@@ -18,6 +18,7 @@ pub enum NUM {
     AnalogComparator      = 0x00007,
     LowLevelDebug         = 0x00008,
     ReadOnlyState         = 0x00009,
+    PrintLog              = 0x0000a,
```

Comment:
This suggests that we plan to maintain both the `Console` and `PrintLog` implementations going forward -- is there any reason for `PrintLog` not to just replace `Console`? Alternatively, we could keep both capsules, but have them use the same driver number. Then board authors can choose whether they need in-order prints, but regardless can use the same userspace libraries

Reply:
Great question -- I think this is worth a discussion in the core WG. The current console driver allows userspace to submit arbitrary length strings and always prints them completely. This driver might truncate them (because it's ultimately relying on copying into a finite in-kernel buffer). 

So I think there are two possible ways forward:
  - Have them as alternative implementations of the same device, with different runtime behavior. In this case, we will want to look at the API to make sure that the API is such that both behaviors are acceptable, and userspace receives enough information (in return values or callbacks) to respond appropriately.
  - Have them as separate devices, with potentially different APIs and behaviors, and have them accessible through different userspace library calls (e.g., printf for console, klog for this one). 

I personally kind of lean towards the second option, but I'm interested to hear other people's thoughts.

Reply:
+1 for discussing this on the core call

Reply:
The resolution on the core call was to go with option #1 (two alternative implementations of the same device/system call API).

Reply:
Does that mean we will also port over the read portion of the console driver to this capsule?

Reply:
I think so, yes.